packages/vow/src/types.js

```diff
@@ -0,0 +1,78 @@
+// @ts-check
+export {};
+
+/**
+ * @template T
+ * @typedef {PromiseLike<T | Vow<T>>} PromiseVow
+ */
+
+/**
+ * @template T
+ * @typedef {import('./E').ERef<T>} ERef
+ */
+
+/**
+ * Creates a type that accepts both near and marshalled references that were
+ * returned from `Remotable` or `Far`, and also promises for such references.
+ *
+ * @template Primary The type of the primary reference.
+ * @template [Local=import('./E').DataOnly<Primary>] The local properties of the object.
+ * @typedef {ERef<Local & import('@endo/eventual-send').RemotableBrand<Local, Primary>>} FarRef
+ */
+
+/**
+ * @template [T=any]
+ * @typedef {object} VowV0 The first version of the vow implementation
+ * object.  CAVEAT: These methods must never be changed or added to, to provide
+ * forward/backward compatibility.  Create a new object and bump its version
+ * number instead.
+ *
+ * @property {() => Promise<T>} shorten Attempt to unwrap all vows in this
+ * promise chain, returning a promise for the final value.  A rejection may
+ * indicate a temporary routing failure requiring a retry, otherwise that the
+ * decider of the terminal promise rejected it.
+ */
+
+/**
+ * @template [T=any]
+ * @typedef {object} VowPayload
+ * @property {import('@endo/eventual-send').FarRef<VowV0<T>>} vowV0
+ */
+
+/**
+ * @template [T=any]
+ * @typedef {import('@endo/pass-style').CopyTagged<
+ *   'Vow', VowPayload<T>
+ * >} Vow
```

Comment:
This structure can't be passed through an endo interface guard.

even `M.any()` fails with `cannot check unrecognized tag "Vow": "[Vow]"`

Reply:
false alarm. the failing pattern was `M.string()`